### DISCUSSION TITLE
[3.9.20 / 4.0.b4] Articles Category Module - Article Field to Order By (Descending Bug)

### DISCUSSION THREAD

**Discussion by LuxLOL** [joomla/joomla-cms#47154] [2. Cold Features]
### Steps to reproduce the issue
I noticed that if i set the option "Article Field to Order By" to "Descending" then it will sort the Articles with the database entries "0000-00-00 00:00:00" first before everything else. The correct way would be to filter this Articles out or sorted then at the very end.

### Expected result

**Sorting Descending:**
Articles -> 2020-01-04 08:08:12
Articles -> 2020-01-03 13:34:52
**OR:**
Articles -> 2020-01-04 08:08:12
Articles -> 2020-01-03 13:34:52
Articles -> 0000-00-00 00:00:00

### Actual result

**Sorting Descending:**
Articles -> 0000-00-00 00:00:00
Articles -> 2020-01-04 08:08:12
Articles -> 2020-01-03 13:34:52

### System information (as much as possible)

PHP v7.4.8 and MySQL v5.5.5-10

### Additional comments

![screen shot 2020-08-06 at 14 09 18](https://issues.joomla.org/uploads/1/4197fb4031e6239549a315b2c46226b1.png)

**Comment by richard67** [comment]
In 3.9.x this might be an issue. But in 4.0? There should not be any old 0000-00-00 dates anymore, they should have real null values instead, or if not nullable have a value not equal to this old null date.

@LuxLOL Have you really tested that on a 4.0 with a new installation or with an updated one from 3.10 (which might have been previously updated from 3.9.x)? I doubt there were really null dates.

**Reply by LuxLOL** [reply]
It looks like it dose not add this 0000-00-00 00:00:00 entries anymore. But it still sets the "Modified Date" if you create a new Article which should only be set if you edit a Article that already exists.

**Comment by LuxLOL** [comment]
There was bug in an extensions that caused this 0000-00-00 problem.

Now Joomla writes the creating date into the modified date when their no date exists.
But this is even worse because now i can't even separate the Updated Articles from the Latest Articles.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30301">issues.joomla.org/tracker/joomla-cms/30301</a>.</sub>
